src/webapp/components/theme-list/ThemePermissionsDialog.tsx

```diff
@@ -16,49 +16,69 @@ export interface ThemePermissionsDialogProps {
     const { themeId, onClose, rows: themes, onChange } = props;
     const { compositionRoot } = useAppContext();
+    const [errors, setErrors] = useState<string[]>([]);
 
     const search = useCallback((query: string) => compositionRoot.users.search(query), [compositionRoot]);
 
     const buildMetaObject = useCallback(
         (id: Id) => {
-            const buildSharings = (type: PermissionType) => {
+            const buildSharings = () => {
                 const theme = themes.find(theme => theme.id === id);
-                return type === "user" ? theme?.sharing?.users : theme?.sharing?.userGroups;
+                return {
+                    users: theme?.sharing?.users,
+                    userGroups: theme?.sharing?.userGroups,
+                    external: theme?.sharing?.external,
+                    public: theme?.sharing?.public,
+                };
             };
 
             return {
                 meta: {
-                    allowPublicAccess: false,
+                    allowPublicAccess: true,
                     allowExternalAccess: false,
                 },
                 object: {
                     id: "",
                     displayName: i18n.t("Access to Themes"),
-                    externalAccess: false,
-                    publicAccess: "",
-                    userAccesses: buildSharings("user"),
-                    userGroupAccesses: buildSharings("userGroup"),
+                    externalAccess: buildSharings().external,
+                    publicAccess: buildSharings().public,
+                    userAccesses: buildSharings().users,
+                    userGroupAccesses: buildSharings().userGroups,
                 },
             };
         },
         [themes]
     );
 
     const onUpdateSharingOptions = useCallback(() => {
-        return async ({ userAccesses: users, userGroupAccesses: userGroups }: ShareUpdate) => {
+        return async ({
+            userAccesses: users,
+            userGroupAccesses: userGroups,
+            publicAccess: publicSharing,
+            externalAccess: external,
+        }: ShareUpdate) => {
             const newThemes = themes.map((theme): Theme => {
-                if (theme.id !== themeId) return theme;
-                return theme.updateSharing({
-                    external: false,
-                    public: "",
-                    users: users ?? [],
-                    userGroups: userGroups ?? [],
-                });
+                if (theme.id !== themeId) {
+                    return theme;
+                } else {
+                    const newTheme = theme.updateSharing({
+                        external: external ?? false,
+                        public: publicSharing ?? "r-------",
+                        users: users ?? [],
+                        userGroups: userGroups ?? [],
+                    });
+                    compositionRoot.themes.save(newTheme).then(errors => setErrors(errors));
```

Comment:
[important] The if/else must be inside the `then`, otherwise it's using the old `errors` state, not the newest one after the save.